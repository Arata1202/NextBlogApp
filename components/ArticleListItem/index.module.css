.list {
  margin-bottom: 20px;
}

.date {
  display: flex;
  justify-content: right;
}

.link {
  display: flex;
  gap: 10px;
}

.image {
  height: auto;
  max-width: 260px;
}

.description {
  font-size: 14.5px;
}

.content {
  max-width: 100%;
}

.title {
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
}

@media (max-width: 1023px) and (min-width: 641px) {
  .image {
    height: auto;
    max-width: 220px;
  }
  .content {
    max-width: 100%;
  }
}

@media (max-width: 640px) {
  .link {
    display: block;
  }
  .image {
    width: 100%;
    max-width: 100%;
    height: auto;
    margin-bottom: 1rem;
  }

  .title {
    font-size: 1.4rem;
  }
}
/* iPadProç”¨ */
/* @media (min-width: 1024px) and (max-width: 1090px) {
  .image {
    width: 210%;
    height: auto;
    max-width: 210%;
  }
} */
