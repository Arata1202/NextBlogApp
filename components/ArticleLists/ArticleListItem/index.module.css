.list {
  margin-bottom: 20px;
}

.date {
  display: flex;
  justify-content: right;
}

.link {
  display: flex;
  gap: 10px;
}

.image {
  height: auto;
  max-width: 300px;
}

.description {
  font-size: 14.5px;
}

.content {
  max-width: 100%;
}

.title {
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
}

@media (max-width: 1023px) and (min-width: 641px) {
  .image {
    height: auto;
    max-width: 300px;
  }
  .content {
    max-width: 100%;
  }
}
@media (max-width: 1081px) and (min-width: 1023px) {
  .image {
    height: auto;
    max-width: 250px;
  }
  .content {
    max-width: 100%;
  }
}

@media (max-width: 717px) and (min-width: 641px) {
  .image {
    height: auto;
    max-width: 200px;
  }
  .content {
    max-width: 100%;
  }
}
@media (max-width: 640px) {
  .link {
    display: block;
  }
  .image {
    width: 100%;
    max-width: 100%;
    height: auto;
    margin-bottom: 1rem;
  }

  .title {
    font-size: 1.4rem;
  }
}
/* iPadProç”¨ */
/* @media (min-width: 1024px) and (max-width: 1090px) {
  .image {
    width: 210%;
    height: auto;
    max-width: 210%;
  }
} */
