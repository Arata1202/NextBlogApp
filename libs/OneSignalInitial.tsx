'use client';

import { useEffect } from 'react';
import OneSignal from 'react-onesignal';

export const OneSignalInitial = () => {
  useEffect(() => {
    const oneSignalInit = async () => {
      await OneSignal.init({
        appId: process.env.NEXT_PUBLIC_ONESIGNAL_APP_ID || '',
        notifyButton: {
          enable: true,
        },
      }).then(() => {
        OneSignal.Slidedown.promptPush();
      });
    };
    oneSignalInit();
  }, []);
  return (
    <div className="mt-5 flex justify-center">
      <div className="onesignal-customlink-container"></div>
    </div>
  );
};
