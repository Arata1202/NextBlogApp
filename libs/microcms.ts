import { createClient } from 'microcms-js-sdk';
import type {
  MicroCMSQueries,
  MicroCMSImage,
  MicroCMSDate,
  MicroCMSContentId,
} from 'microcms-js-sdk';
import { notFound } from 'next/navigation';

// カテゴリーの型定義
export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// タグの型定義
export type Tag2 = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// ライターの型定義
export type Writer = {
  name: string;
  profile: string;
  image?: MicroCMSImage;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type ContentBlock = {
  adsense?: string;
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type RelatedArticle = {
  articleLink3?: string;
};

// ブログの型定義
export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  tags?: Tag[];
  tags2?: Tag2[];
  content_blocks: ContentBlock[];
  introduction_blocks: IntroductionBlock[];
  related_articles?: RelatedArticle[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

// Initialize Client SDK.
export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

// ブログ一覧を取得
export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);
  return listData;
};

// ブログの詳細を取得
export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// カテゴリーの一覧を取得
export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

// タグの一覧を取得
export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag2>({
      endpoint: 'tags2',
      queries,
    })
    .catch(notFound);

  return listData;
};

// カテゴリーの詳細を取得
export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// タグの詳細を取得
export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag2>({
      endpoint: 'tags2',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
