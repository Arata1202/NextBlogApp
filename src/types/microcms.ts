import type { MicroCMSImage, MicroCMSDate, MicroCMSContentId } from 'microcms-js-sdk';

export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  categories: Category[];
  tags?: Tag[];
  introduction_blocks: IntroductionBlock[];
  content_blocks: ContentBlock[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

export type Category = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};

export type ContentBlock = {
  google_adsense?: string;
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
