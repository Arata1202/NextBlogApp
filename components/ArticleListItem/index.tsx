'use client';

import Link from 'next/link';
import Image from 'next/image';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';
import TagList from '../TagList';
import PublishedDate from '../Date';

type Props = {
  article: Article;
};

export default function ArticleListItem({ article }: Props) {
  return (
    <li className={styles.list}>
      <Link
        href={`/articles/${article.id}`}
        className={`${styles.link} p-5 border border-gray-300 rounded-2xl`}
      >
        {article.thumbnail ? (
          <picture>
            <source
              type="image/webp"
              media="(max-width: 640px)"
              srcSet={`${article.thumbnail?.url}?fm=webp&w=414 1x, ${article.thumbnail?.url}?fm=webp&w=414&dpr=2 2x`}
            />
            <source
              type="image/webp"
              srcSet={`${article.thumbnail?.url}?fm=webp&fit=crop&w=240&h=126 1x, ${article.thumbnail?.url}?fm=webp&fit=crop&w=240&h=126&dpr=2 2x`}
            />
            <img
              src={article.thumbnail?.url || `/noimage.png`}
              alt=""
              className={styles.image}
              width={article.thumbnail?.width}
              height={article.thumbnail?.height}
            />
          </picture>
        ) : (
          <Image
            className={styles.image}
            src="/no-image.png"
            alt="No Image"
            width={1200}
            height={630}
          />
        )}
        <dl className={styles.content}>
          <dt className={styles.title}>{article.title}</dt>
          {/* <dt className={styles.description}>{article.description}</dt> */}
          <dd>
            <TagList tags={article.tags} hasLink={true} />
          </dd>
          <dd className={styles.date}>
            <PublishedDate date={article.publishedAt || article.createdAt} />
          </dd>
        </dl>
      </Link>
    </li>
  );
}
