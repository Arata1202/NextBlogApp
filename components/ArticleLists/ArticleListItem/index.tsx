'use client';

import React from 'react';
import { useTheme } from 'next-themes';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';
import PublishedDate from '../../Elements/Date';

type Props = {
  article: Article;
};

export default function ArticleListItem({ article }: Props) {
  const { theme } = useTheme();

  const isNextDayOrLater = (date1: string, date2: string) => {
    const d1 = new Date(date1);
    const d2 = new Date(date2);
    d1.setHours(0, 0, 0, 0);
    d2.setHours(0, 0, 0, 0);
    return d1 > d2;
  };

  return (
    <>
      <li className={styles.list}>
        <a
          href={`/articles/${article.id}`}
          className={`${styles.link} p-2 border shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
        >
          <picture>
            <source
              type="image/webp"
              media="(max-width: 640px)"
              srcSet={`${article.thumbnail.url}?fm=webp&w=414 1x, ${article.thumbnail.url}?fm=webp&w=414&dpr=2 2x`}
            />
            <source
              type="image/webp"
              srcSet={`${article.thumbnail.url}?fm=webp&fit=crop&w=240&h=126 1x, ${article.thumbnail.url}?fm=webp&fit=crop&w=240&h=126&dpr=2 2x`}
            />
            <img
              src={article.thumbnail.url}
              alt="サムネイル"
              className={styles.image}
              width={article.thumbnail.width}
              height={article.thumbnail.height}
            />
          </picture>
          <div className={styles.content}>
            <div className={styles.title}>{article.title}</div>
            <div className={styles.description}>{article.description}</div>
            <div className={styles.date}>
              <PublishedDate date={article.publishedAt!} updatedAt={false} />
              {article.updatedAt && isNextDayOrLater(article.updatedAt, article.publishedAt!) && (
                <>
                  &nbsp;&nbsp;&nbsp;&nbsp;
                  <PublishedDate date={article.updatedAt!} updatedAt={true} />
                </>
              )}
            </div>
          </div>
        </a>
      </li>
    </>
  );
}
