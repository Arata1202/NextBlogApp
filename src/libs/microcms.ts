import { notFound } from 'next/navigation';
import { createClient } from 'microcms-js-sdk';
import type { MicroCMSQueries } from 'microcms-js-sdk';
import { Blog, Category, Tag } from '@/types/microcms';

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

export const getAllLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllCategoryLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Category>({
      endpoint: 'categories',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllTagLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
