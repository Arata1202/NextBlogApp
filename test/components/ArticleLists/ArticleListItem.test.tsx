import { render } from '@testing-library/react';
import ArticleListItem from '@/components/ArticleLists/ArticleListItem';
import { Article } from '@/libs/microcms';

jest.mock('@/components/Categories/CategoryList');
jest.mock('@/components/Elements/Date');

const article: Article = {
  id: '1',
  title: 'タイトル',
  description: '概要',
  thumbnail: { url: 'https://thumbnail.com/thumbnail.jpg' },
  publishedAt: '2023-10-30T12:00:00Z',
  content_blocks: [],
  introduction_blocks: [],
  createdAt: '2023-10-30T12:00:00Z',
  updatedAt: '2023-10-30T12:00:00Z',
};

describe('ArticleListItem', () => {
  test('スナップショット(ArticleListItem)', () => {
    const { asFragment } = render(<ArticleListItem key={article.id} article={article} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
