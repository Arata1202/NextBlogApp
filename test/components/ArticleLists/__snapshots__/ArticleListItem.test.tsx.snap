// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem スナップショット(ArticleListItem) 1`] = `
<DocumentFragment>
  <li
    class="list"
  >
    <a
      class="link p-2 border border-gray-300 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1"
      href="/articles/1"
    >
      <picture>
        <source
          media="(max-width: 640px)"
          srcset="https://thumbnail.com/thumbnail.jpg?fm=webp&w=414 1x, https://thumbnail.com/thumbnail.jpg?fm=webp&w=414&dpr=2 2x"
          type="image/webp"
        />
        <source
          srcset="https://thumbnail.com/thumbnail.jpg?fm=webp&fit=crop&w=240&h=126 1x, https://thumbnail.com/thumbnail.jpg?fm=webp&fit=crop&w=240&h=126&dpr=2 2x"
          type="image/webp"
        />
        <img
          alt="サムネイル"
          class="image"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="600"
        />
      </picture>
      <div
        class="content"
      >
        <div
          class="title"
        >
          タイトル
        </div>
        <div
          class="description"
        >
          概要
        </div>
        <div
          class="date"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5 mr-2 mt-4"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.25 12.75V12A2.25 2.25 0 0 1 4.5 9.75h15A2.25 2.25 0 0 1 21.75 12v.75m-8.69-6.44-2.12-2.12a1.5 1.5 0 0 0-1.061-.44H4.5A2.25 2.25 0 0 0 2.25 6v12a2.25 2.25 0 0 0 2.25 2.25h15A2.25 2.25 0 0 0 21.75 18V9a2.25 2.25 0 0 0-2.25-2.25h-5.379a1.5 1.5 0 0 1-1.06-.44Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
              
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
